### ISSUE TITLE
[sign_in_with_apple] On Android, the app is restarting after the callback intent from oauth redirect is received

### ISSUE THREAD

**Issue by eduribas** [aboutyou/dart_packages#81]
On Android, after calling `await SignInWithApple.getAppleIDCredential()` and performing the authentication on the chrome tab, the call returns with the credential but the app is restarted. If I put a breakpoint on `runApp(MyApp());` I can see that it's called again. Is that the expected behavior?

I was able to reproduce that using the example app.

**Comment by HenriBeck** [comment]
Hey @eduribas, I will have a look at this tomorrow.

Could you please post the output of `flutter doctor -v` so I can know on which versions you are?

**Comment by eduribas** [comment]
Hi @HenriBeck , I tracked down the problem and sent a PR that seems to solve the problem on my case. This is my flutter doctor -v:
```
[✓] Flutter (Channel stable, v1.17.1, on Mac OS X 10.15.4 19E287, locale en-BR)
    • Flutter version 1.17.1 at /Users/eduribas/Dev/Library/flutter
    • Framework revision f7a6a7906b (10 days ago), 2020-05-12 18:39:00 -0700
    • Engine revision 6bc433c6b6
    • Dart version 2.8.2

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.1)
    • Android SDK at /Users/eduribas/Library/Android/sdk
    • Platform android-29, build-tools 29.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.4.1, Build version 11E503a
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] VS Code (version 1.45.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.10.2

[✓] Connected device (1 available)
    • SM G975F • RQ8M20K40ZF • android-arm64 • Android 10 (API 29)

• No issues found!
```

**Comment by tp** [comment]
That behavior was not intended explicitly and definitely not ideal.

Thanks for contributing the fix @eduribas, it's released as version 2.3.0.
